### ISSUE TITLE
07-naive-bayes-playstore-reviews.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/GuilloMansa-MachineLearning#6]
Hola @GuilloMansa ""

---

## **LO QUE HAS HECHO BIEN**

-  **Eliminación de `package_name`**: Correcto, esta variable no aporta información para el análisis de sentimientos.
- **Preprocesamiento de texto**: Muy bien aplicado el `.str.strip().str.lower()` para normalizar el texto.
- **Entrenamiento de los 3 modelos**: Probaste correctamente MultinomialNB, GaussianNB y BernoulliNB.
- **Observación final sobre MultinomialNB**: Correcta tu conclusión de que MultinomialNB es el mejor para este tipo de datos.

---

### **1. Hiperparámetros con modelo equivocado**

```python
random_search = RandomizedSearchCV(model, hyperparams, ...)
```

**Problema**: La variable `model` en este punto contiene `BernoulliNB()` (el último modelo entrenado), pero tú concluiste que **MultinomialNB** era el mejor. Estás optimizando el modelo equivocado.

**Solución**: Deberías crear explícitamente el modelo correcto:
```python
model_multi = MultinomialNB()
random_search = RandomizedSearchCV(model_multi, hyperparams, ...)
```

---

### **2. PASO 6 NO COMPLETADO - Explorar alternativas**

El enunciado te pedía:
> *"¿Qué otros modelos de los que hemos estudiado podrías utilizar para intentar superar los resultados de un Naive Bayes? Arguméntalo y entrena el modelo."*

**No has hecho este paso**. Podrías haber probado:
- **Regresión Logística**: Funciona muy bien con texto vectorizado y es muy interpretable.
- **SVM (Support Vector Machine)**: Excelente para clasificación de texto.
- **Random Forest**: Como mencionaba el enunciado (aunque el enunciado confunde esto con optimización).

---

### **8. El enunciado dice "usar con Random Forest"**

El Paso 4 dice textualmente:
> *"trata de optimizar sus resultados con un random forest"*

Esto está **mal redactado en el enunciado** (confunde optimización de hiperparámetros con comparación de modelos). Pero tú tampoco lo has intentado. Puedes entrenar un RandomForestClassifier para comparar.

---

### **9. No muestras los mejores parámetros encontrados**

Después del GridSearchCV, deberías mostrar:
```python
Mejores parámetros: {grid_search_multi.best_params_}"
Mejor score: {grid_search_multi.best_score_}")
```

---

### **10. Métricas insuficientes**

Solo usas `accuracy_score`. Para un problema de clasificación binaria de sentimientos, sería importante añadir:
- **Classification Report**: precision, recall, f1-score
- **Matriz de confusión**: para ver falsos positivos/negativos

```python
from sklearn.metrics import classification_report, confusion_matrix
print(classification_report(y_test, y_pred_test))
```
